### PR TITLE
Add rnp_key_add_uid function

### PR DESCRIPTION
**Pull request by randombit**

### CODE REVIEW

File: src/lib/rnp2.c [hunk 1]
```diff
@@ -1999,6 +2000,44 @@ key_get_uid_at(pgp_key_t *key, size_t idx, char **uid)
...
+                bool             primary)
+{
+    rnp_selfsig_cert_info info;
+    pgp_hash_alg_t        hash_alg = PGP_HASH_UNKNOWN;
+    pgp_key_t *           key = get_key_prefer_public(handle);
+    pgp_key_t *           seckey = get_key_require_secret(handle);
+
+    memset(&info, 0, sizeof(info));
+
+    if (!key || !seckey)
```
Comment (henrydcase) [inline src/lib/rnp2.c]: I would add !uid as you call strlen() later on it

File: include/rnp/rnp2.h [hunk 2]
```diff
@@ -237,6 +237,13 @@ rnp_result_t rnp_key_get_primary_uid(rnp_key_handle_t key, char **uid);
 rnp_result_t rnp_key_get_uid_count(rnp_key_handle_t key, size_t *count);
 rnp_result_t rnp_key_get_uid_at(rnp_key_handle_t key, size_t idx, char **uid);
 
+rnp_result_t rnp_key_add_uid(rnp_key_handle_t key,
```
Comment (henrydcase) [inline include/rnp/rnp2.h]: As that's public API some comments could be helpful

File: src/lib/rnp2.c [hunk 3]
```diff
@@ -1999,6 +2000,44 @@ key_get_uid_at(pgp_key_t *key, size_t idx, char **uid)
...
+        return RNP_ERROR_BAD_PARAMETERS;
+
+    ARRAY_LOOKUP_BY_STRCASE(hash_alg_map, string, type, hash, hash_alg);
+    if (hash_alg == PGP_HASH_UNKNOWN) {
+        return RNP_ERROR_BAD_FORMAT;
```
Comment (henrydcase) [inline src/lib/rnp2.c]: shouldn't the error code be RNP_ERROR_BAD_PARAMETERS?

Comment (randombit) [reply src/lib/rnp2.c]: BAD_FORMAT is used in other similar cases (eg `rnp_op_encrypt_add_password`) and I probably just copied it from there. I agree BAD_PARAM makes more sense here though.

File: src/lib/rnp2.c (hunk 1 above)
Comment (randombit) [reply src/lib/rnp2.c]: Fixed

File: include/rnp/rnp2.h (hunk 2 above)
Comment (randombit) [reply include/rnp/rnp2.h]: Done

**Pull request merged**